### DISCUSSION TITLE
creating buffer info without initial attrib values

### DISCUSSION THREAD

**Discussion by Vadi** [greggman/twgl.js#201] [Q&A] [answered]
I didn't know discussions exist in Github.

I'm trying to create a `bufferInfo` object without any initial data. Below one works, passing empty array for `data` does not throw error nor works.

This one works:

```js
  const arrays = {
    position: {numComponents:2,  data: [
      0, 0,
      10, 10,
      0, 10,
    ]},
    indices: [
      0, 1, 2,
    ]
  };

  const bufferInfo = twgl.createBufferInfoFromArrays(gl, arrays);
```

Passing empty array,

```js
  const arrays = {
    position: {numComponents:2,  data: []},
    indices: []
  };

  const bufferInfo = twgl.createBufferInfoFromArrays(gl, arrays);

// .. later in the code, during render

    twgl.setAttribInfoBufferFromArray(
      gl,
      bufferInfo.attribs.position,
      new Float32Array([tri[0], tri[1], tri[2], tri[3], tri[4], tri[5]])
    );
```

I don't see anything getting rendererd.

QUESTION: Is this valid approach? Or is there a way to setup buffer info during initialization phase.

Thanks!

**Comment by greggman** [comment, marked as answer]
It's not really clear what you're trying to do. For now you could do this

```
  const arrays = {
    position: {numComponents:2,  data: new Float32Array(6)},
    indices: new Uint16Array(numIndices),
  };
```

twgl does not provide a way to update indices. For that you'll have to use raw WebGL.

**Reply by Vadi** [reply]
You answered my question. I was passing empty arrays to `data` it doesn't work.
